packages/engine/Source/Scene/Scene.js

```diff
@@ -3774,11 +3770,12 @@ Scene.prototype.updateHeight = function (
   );
   const removeRemovedListener = this.primitives.primitiveRemoved.addEventListener(
     (primitive) => {
-      if (!primitive.isCesium3DTileset) {
+      if (primitive.isDestroyed() || !primitive.isCesium3DTileset) {
         return;
       }
-
-      tilesetRemoveCallbacks[primitive.id]();
+      if (defined(tilesetRemoveCallbacks[primitive.id])) {
+        tilesetRemoveCallbacks[primitive.id]();
+      }
```

Comment:
No change needed on this right now but I did want to point out that [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) _does_ work on functions which would account for this. I'm still partly undecided if it's easier or harder to read though. Mostly just commenting for awareness if you didn't know

```suggestion
        tilesetRemoveCallbacks[primitive.id]?.();
```